### ISSUE TITLE
just after a few min get this error

### ISSUE THREAD

**Issue by Slepperpon** [pdlove/homebridge-smartthings#40]
/usr/local/lib/node_modules/homebridge-smartthings/index.js:70
            } else if (myList.error) {
                             ^

TypeError: Cannot read property 'error' of undefined
    at /usr/local/lib/node_modules/homebridge-smartthings/index.js:70:30
    at /usr/local/lib/node_modules/homebridge-smartthings/lib/smartthingsapi.js:93:13
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-smartthings/lib/smartthingsapi.js:46:27)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:893:12)
    at doNTCallback2 (node.js:429:9)
    at process._tickCallback (node.js:343:17)

**Comment by tannebil** [comment]
Got the same thing.

**Comment by pdlove** [comment]
I got this error once yesterday while I was at work and assumed it was an issue with my new version of the plugin. I determined after that the issue is that it doesn't make sure a valid reply came back from your SmartApp before checking the error. You'll see a correction this evening.

**Comment by Slepperpon** [comment]
error at req:  { [Error: getaddrinfo ENOTFOUND graph.api.smartthings.com graph.api.smartthings.com:443]
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'graph.api.smartthings.com',
  host: 'graph.api.smartthings.com',
  port: '443' }
error at req:  { [Error: getaddrinfo ENOTFOUND graph.api.smartthings.com graph.api.smartthings.com:443]
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'graph.api.smartthings.com',
  host: 'graph.api.smartthings.com',
  port: '443' }
error at req:  { [Error: getaddrinfo ENOTFOUND graph.api.smartthings.com graph.api.smartthings.com:443]
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'graph.api.smartthings.com',
  host: 'graph.api.smartthings.com',
  port: '443' }
error at req:  { [Error: getaddrinfo ENOTFOUND graph.api.smartthings.com graph.api.smartthings.com:443]
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'graph.api.smartthings.com',
  host: 'graph.api.smartthings.com',
  port: '443' }
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::54829
    at Object.exports._errnoException (util.js:837:11)
    at exports._exceptionWithHostPort (util.js:860:20)
    at Server._listen2 (net.js:1231:14)
    at listen (net.js:1267:10)
    at Server.listen (net.js:1363:5)
    at new EventedHTTPServerConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/eventedhttp.js:125:20)
    at EventedHTTPServer._onConnection (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/eventedhttp.js:79:20)
    at emitOne (events.js:77:13)
    at Server.emit (events.js:169:7)
    at TCP.onconnection (net.js:1427:8)

**Comment by pdlove** [comment]
Eaddrinuse is from launching it twice some how... The other one looks to be a dns issue. Can you verify your API is on the US server and try to go there from the web browser?

**Comment by Slepperpon** [comment]
It was running for hours before that message came up. And as far as the api you are talking about smarthings hub correct? It is on a us server yes

Date: Wed, 17 Feb 2016 07:03:03 -0800

**Comment by pdlove** [comment]
Im getting ENOTFOUND  on occasion... twice in the last 3 days.

**Comment by rickybloomfield** [comment]
I got ENOTFOUND a couple times as well. Not since upgrading to latest, though (yesterday). I thought it was initially due to another homebridge interface, but now I'm not sure. I'll see how it goes. Before the latest flurry of changes it had been pretty stable for me. Recently, though, it will konk out after several hours of use. Hopefully the latest changes will help with that.

**Comment by rickybloomfield** [comment]
In addition to the ENOTFOUND error, I just started getting a bunch of errors like this:

error at req:  { [Error: connect ETIMEDOUT 174.129.9.165:443]
  code: 'ETIMEDOUT',
  errno: 'ETIMEDOUT',
  syscall: 'connect',
  address: '174.129.9.165',
  port: 443 }

**Comment by pdlove** [comment]
#45 #46 #47 
Please try the latest release. I've tweaked the timing values and can't seem to get my setup to break.
Because I tweaked defaults only, verify you don't have polling_seconds in config.json set higher than 60. I tested it at 60 on my setup without losing any devices after several hours.

**Comment by pdlove** [comment]
When I left home and then returned, I lost several devices. My tablet I left at home did not. Anyone else experience this?

**Comment by nguyenshane** [comment]
I lost all of 19 devices twice today. First time was right when I left home. Second time was just about 15 minutes after I restart homebridge, remove persist folder and reconnect all devices.

**Comment by Slepperpon** [comment]
Still running after 5 days no problems. All devices still there with remote access as well. 

Adam Hoffman
District Team Lead
District 9445
NTB 892

**Comment by coolcom** [comment]
Everything running fine for a couple days, and then I got this error, and cannot relaunch without getting this.
```
bed-htpc:~ htpc$ homebridge 
Loaded plugin: homebridge-smartthings
Registering platform 'homebridge-smartthings.SmartThings'
---
Loaded config.json with 0 accessories and 1 platforms.
---
Loading 1 platforms...
[SmartThings] Initializing SmartThings platform...
[SmartThings] Fetching Smart Things devices.
/usr/local/lib/node_modules/homebridge-smartthings/index.js:76
            } else if (myList.error) {
                             ^

TypeError: Cannot read property 'error' of undefined
    at /usr/local/lib/node_modules/homebridge-smartthings/index.js:76:30
    at /usr/local/lib/node_modules/homebridge-smartthings/lib/smartthingsapi.js:93:13
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-smartthings/lib/smartthingsapi.js:46:27)
    at emitNone (events.js:85:20)
    at IncomingMessage.emit (events.js:179:7)
    at endReadableNT (_stream_readable.js:913:12)
    at _combinedTickCallback (node.js:377:13)
    at process._tickCallback (node.js:401:11)
```

**Comment by Slepperpon** [comment]
Same as coolcom working for 10 days then crashed and will not start back up

**Comment by coolcom** [comment]
Update- I just tried restarting Homebridge after it had been off for ~2 hours, and it started up fine.

**Comment by pdlove** [comment]
@slepperpon  please confirm the version of the plug-in and that your API keys are still the same.  I've heard of at least one report where the API token changed.

**Comment by Slepperpon** [comment]
Made no changes and it works just fine this morning.

Date: Thu, 25 Feb 2016 15:33:56 -0800

**Comment by pdlove** [comment]
Try this with plugin version 0.4.4. I've corrected a lot on the error handling. You'll still get the connection errors because that is all smart things being an issue.... but it shouldn't cause home bridge to crash as it has been doing.

**Comment by pdlove** [comment]
I'm moving all discussion to issue #58. I was able to iron out one major issue last night an just released that as 0.4.6.
